### ISSUE TITLE
Should the trailer have nlinks=1 for CRC?

### ISSUE THREAD

**Issue by grahamc** [jcreekmore/cpio-rs#6]
I'm comparing the CPIO's I'm generating with this library and the GNU `cpio` tool. The only difference I'm seeing is in the trailer:

```diff
  ino	mode	uid	gid	nlink	mtime	bytes	devmaj	devmin	rdevmaj	rdevmin	trailer	hash	name
- 0	0	0	0	1	0	0	0	0	0	0	true	47DEQpj8HBSa+/TImW+5JCeuQeRkm5NMpJWZG3hSuFU=	TRAILER!!!
+ 0	0	0	0	0	0	0	0	0	0	0	true	47DEQpj8HBSa+/TImW+5JCeuQeRkm5NMpJWZG3hSuFU=	TRAILER!!!
```

This was interesting to me, so I dug into the source code of the cpio utility. It says the nlink must be 1 for the crc format. I don't understand why, nor do I see any docs about it. Also, the commit adding that line is the first commit on the repo:

```
a783e2c1 (Sergey Poznyakoff 2004-02-27 13:28:40 +0000 841)   file_hdr.c_ino = 0;
a783e2c1 (Sergey Poznyakoff 2004-02-27 13:28:40 +0000 842)   file_hdr.c_mode = 0;
a783e2c1 (Sergey Poznyakoff 2004-02-27 13:28:40 +0000 843)   file_hdr.c_uid = 0;
a783e2c1 (Sergey Poznyakoff 2004-02-27 13:28:40 +0000 844)   file_hdr.c_gid = 0;
a783e2c1 (Sergey Poznyakoff 2004-02-27 13:28:40 +0000 845)   file_hdr.c_nlink = 1;              /* Must be 1 for crc format.  */
a783e2c1 (Sergey Poznyakoff 2004-02-27 13:28:40 +0000 846)   file_hdr.c_dev_maj = 0;
a783e2c1 (Sergey Poznyakoff 2004-02-27 13:28:40 +0000 847)   file_hdr.c_dev_min = 0;
a783e2c1 (Sergey Poznyakoff 2004-02-27 13:28:40 +0000 848)   file_hdr.c_rdev_maj = 0;
a783e2c1 (Sergey Poznyakoff 2004-02-27 13:28:40 +0000 849)   file_hdr.c_rdev_min = 0;
a783e2c1 (Sergey Poznyakoff 2004-02-27 13:28:40 +0000 850)   file_hdr.c_mtime = 0;
a783e2c1 (Sergey Poznyakoff 2004-02-27 13:28:40 +0000 851)   file_hdr.c_chksum = 0;
a783e2c1 (Sergey Poznyakoff 2004-02-27 13:28:40 +0000 852) 
a783e2c1 (Sergey Poznyakoff 2004-02-27 13:28:40 +0000 853)   file_hdr.c_filesize = 0;
d36ec5f4 (Pavel Raiskup     2016-01-26 23:17:54 +0100 854)   cpio_set_c_name (&file_hdr, CPIO_TRAILER_NAME);
```

So, I wonder if this library should follow suit. Do you have any opinions?

**Comment by jcreekmore** [comment]
Seems reasonable to me. I purposefully didn’t look at GNU cpio source and, if I remember correctly, wrote this using the description of the file format on Wikipedia and then figuring out what didn’t work when I was wanting to use this to manipulate Linux initramfs images. So while the GNU cpio code says it is required, it didn’t seem to matter in my use case at least.

**Comment by grahamc** [comment]
Thank you!

**Comment by grahamc** [comment]
By the way I don't want to impose, but would you mind publishing a release to crates.io?

**Comment by jcreekmore** [comment]
Released 0.2.2

**Comment by grahamc** [comment]
Thank you!
